md/runtime-monitoring.md

```diff
@@ -0,0 +1,102 @@
+It executes in a Java thread.
+
+**Connector work**: a unit piece of code that specifically executes Bonita connectors
+
+**Work queue**: a queue storing pending works before they are taken by threads for execution 
+
+## Bonita-related metrics
+Bonita-related metrics are **enabled by default** and cannot be disabled. Here they are:
+* The number of currently running works, under the logical key name **org.bonitasoft.engine.work.works.running**
+* The number of currently pending works, waiting in the work queue to be treated, under the logical key name **org.bonitasoft.engine.work.works.pending**
+* The total number of executed works (since the last start of Bonita platform), under the logical key name **org.bonitasoft.engine.work.works.executed**
+* The number of currently running connector works, under the logical key name **org.bonitasoft.engine.connector.connectors.running**
+* The number of currently pending connector works, waiting in the connector work queue to be treated,
+under the logical key name **org.bonitasoft.engine.connector.connectors.pending**
+* The total number of executed connector works (since the last start of Bonita platform), under the logical key name **org.bonitasoft.engine.connector.connectors.executed**
+
+
+
+## Activating specific monitoring metrics
+
+Retrieve current configuration by running:
+```bash
+./setup/setup.sh pull
+```
+and edit file `./setup/platform_conf/current/platform_engine/bonita-platform-community-custom.properties`.  
+You will see, in the `# Monitoring` section, two series of properties with their default value:
+
+    ## Monitoring
+    ## publish metrics to JMX:
+    #org.bonitasoft.engine.monitoring.jmx.enable=true
+    ## periodically print metrics to logs (bonita-related metrics only):
+    #org.bonitasoft.engine.monitoring.logging.enable=false
+    ## print to logs every minute by default (in the ISO-8601 duration format):
+    #org.bonitasoft.engine.monitoring.logging.step=PT1M
+    #
+    ## publish metrics related to JVM memory:
+    #org.bonitasoft.engine.monitoring.metrics.jvm.memory.enable=false
+    ## publish metrics related to JVM Threads:
+    #org.bonitasoft.engine.monitoring.metrics.jvm.threads.enable=false
+    ## publish metrics related to JVM garbage collection:
+    #org.bonitasoft.engine.monitoring.metrics.jvm.gc.enable=false
+    ## publish technical metrics related to Worker / Connector thread pools:
+    #org.bonitasoft.engine.monitoring.metrics.executors.enable=false
+
+The first series is the list of **publishers** that can be activated.  
+In Community edition, 2 publishers are provided:
+* JMX (enabled by default), that allows to use any JMX console to monitor your favorite metrics (except JVM metrics)
+* Logging (disabled by default), that prints to standard Bonita log file the Bonita-related metrics. Print interval can
+be changed (property `org.bonitasoft.engine.monitoring.logging.step`).
```

Comment:
Does it only change the print interval or also the "publish interval", so it can also be used to manage the accuracy of a graph, stuff like that?

Reply:
logging publisher only prints to bonita logs (bonita-2013-09-11.log). It does not really "publish" to some external system.

Reply:
Ok, so I guess it's the external system that asks Bonita "Hey, give me your last value for the metric ***" with its own interval ?

Reply:
there is publish interval but pull interval from ext system (prometheus conf). When requested it provides current value of metrics (no concept of last value but instant one)

Reply:
Just to clarify, the parameter to specify to **print interval** is only for the Bonita logging publisher.
Other publishers have their own mechanism.

Reply:
step configuration is available on multiple registries:
* `prometheus`:
```
The step size to use in computing windowed statistics like max. The default is 1 minute.
To get the most out of these statistics, align the step interval to be close to your scrape interval.
```
* `jmx`:
```
The step size (reporting frequency, max decaying) to use. The default is 1 minute.
```
* `logging`:
```
The step size (reporting frequency) to use. The default is 1 minute.
```

for logging it represent each time we push metrics to logs.
for jmx and prometheus it represent at what interval we get new value (not for all metrics)

see doc on micrometer: https://micrometer.io/docs/concepts#rate-aggregation